src/Microsoft.AspNet.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs

```diff
@@ -208,5 +210,14 @@ public override IEnumerator<string> GetEnumerator()
             // can use this to iterate through the values of the buffer.
             return _buffer.GetEnumerator();
         }
+
+        /// <inheritdoc />
+        public override void WriteTo(TextWriter writer, IHtmlEncoder encoder)
+        {
+            foreach (var entry in _buffer)
+            {
+                writer.Write(entry);
```

Comment:
Currently, this function receives encoded content only. I've an upcoming PR where I delay the encoding till the content is written.